src/components/dashboard/dashboard.service.ts

```diff
@@ -697,7 +697,7 @@ export class DashboardService implements OnDestroy {
 
         // find the positions required
         const targetSpaces = this.getOccupiedSpaces().filter(space => space.widget === widget).map(space => {
-            return { column: space.column - widget.getColumnSpan(), row: space.row, widget: space.widget };
+            return { column: space.column - this._actionWidget.widget.getColumnSpan(), row: space.row, widget: space.widget };
```

Comment:
I don't think this change is appropriate - see the `if` statement above. The function is intended to work on the parameter provided.

Reply:
Hi Donna, the reason I included this is if there is another widget placed beside the one being moved it causes it to moved right a bit too far. See this video for a look at the issue - https://gyazo.com/cbd43aa31d52dfe933672f2670fbfb5f

From my investigation it seemed targetSpaces is where the widget is planning to move right. Without this, it was being calculated to be columns 4-7 (this is why the alert widget moves further than necessary) but in reality the widget should only move to occupy the width of the action widget ie columns 2-5. I hope this explains the thought process behind this!

Reply:
That's fair. I noticed this issue, which I assumed was due to this change, but perhaps I misunderstood:
[![Image from Gyazo](https://i.gyazo.com/93a4f21cbe7ffedb72ff14650e31b00c.gif)](https://gyazo.com/93a4f21cbe7ffedb72ff14650e31b00c)